packages/aws-cdk/lib/cdk-toolkit.ts

```diff
@@ -268,7 +291,31 @@ export class CdkToolkit {
         }
       }
       print('\n✨  Total time: %ss\n', formatTime(elapsedSynthTime + elapsedDeployTime));
-    }
+    };
+
+    const isStackUnblocked = (stack: cxapi.CloudFormationStackArtifact) =>
+      stack.dependencies
+        .map(({ id }) => id)
+        .filter((id) => !id.endsWith('.assets'))
+        .every((id) => !stacksAwaitingDeploy[id]);
+
+    const enqueueStackDeploys = async () => {
+      await Promise.all(stacks.map(async (stack) => {
+        if (isStackUnblocked(stack)) {
+          // Find current index due to stacks list changing within loop
+          const index = stacks.indexOf(stack);
+          stacks.splice(index, 1);
+
+          await queue.add(async () => {
```

Comment:
@rix0rrr I think we need to collect all the queue'd promises to await with the `Promise.all`. Awaiting here will block parallel queueing

```suggestion
          return queue.add(async () => {
```